lib/cartopy/mpl/patch.py

```diff
@@ -302,79 +322,83 @@ def path_to_shapely(path, force_ccw=False):
                                                     axis=1)
 
         if all(verts_same_as_first):
-            geom = sgeom.Point(path_verts[0, :])
-        elif path_verts.shape[0] > 4 and path_codes[-1] == Path.CLOSEPOLY:
-            geom = sgeom.Polygon(path_verts[:-1, :])
+            points.append(sgeom.Point(path_verts[0, :]))
+        elif not(path_verts.shape[0] > 4 and path_codes[-1] == Path.CLOSEPOLY):
+            linestrings.append(sgeom.LineString(path_verts))
         else:
-            geom = sgeom.LineString(path_verts)
-
-        # If geom is a Polygon and is contained within the last geom in
-        # collection, it usually needs to be an interior to that geom (e.g. a
-        # lake within a land mass).  Sometimes there is a further geom within
-        # this interior (e.g. an island in a lake, or some instances of
-        # contours).  This needs to be a new external geom in the collection.
-        if geom.is_empty:
-            pass
```

Comment:
These lines were introduced at #560, but the test added then has since been updated to expect [a list of points](https://github.com/SciTools/cartopy/blob/f8fee7ff052e6ca7672b4e85edc15014009a08d1/lib/cartopy/tests/mpl/test_patch.py#L26).  So I don't think this loop is relevant any more.